### PR TITLE
Improve zsh completion

### PR DESCRIPTION
**Pull request by Freed-Wu**
Refer https://github.com/pallets/click/pull/2544/files

### CODE REVIEW

**Comment by Freed-Wu** [comment]
@wtsnjp

File: doc/texdoc.tex [hunk 1]
```diff
@@ -787,13 +787,7 @@ \subsection{Zsh}
...
   zsh configuration file.
 
-\item adding a file somewhere in \code{\$fpath} with the following contents:
-%
-\begin{htcode}
-#compdef texdoc
-(( $+functions[__texdoc] )) || eval "$(texdoc \textbackslash
-  --print-completion zsh)" && __texdoc
-\end{htcode}
+\item creating a file named \code{_texdoc} in \code{\$fpath} whose content is the output of \code{texdoc \lopt{print-completion} zsh}.
```
Comment (wtsnjp) [inline doc/texdoc.tex:790]: We need an escape backslash for `_` here, so `\code{\_texdoc}`, otherwise the compilation of this document has failed

**Review by wtsnjp — CHANGES_REQUESTED** [review]

File: doc/texdoc.tex (hunk 1 above)
Comment (Freed-Wu) [reply doc/texdoc.tex]: Fixed.

**Comment by wtsnjp** [comment]
LGTM, thanks.

**Pull request merged**